### PR TITLE
chore: backport release/v4.3.x commits to release/v4.4.x

### PR DESCRIPTION
**Pull request by RafilxTenfen**
## Summary

Brings `release/v4.4.x` up to parity with `release/v4.3.x` by cherry-picking every commit that exists on `release/v4.3.x` but not here, then layering follow-ups that keep the v4.4 upgrade-handler layout consistent with what actually shipped on mainnet.

## What's in the branch

**12 workflow / CHANGELOG cherry-picks** (empty after `--ours` conflict resolution because `release/v4.4.x` already has equivalent backports under different PR numbers):
PRs #1922, #1928, #1932, #1935, #1943, #1944, #1948, #1977, #1986, #1999, #2002, #2004.

**2 security cherry-picks:**

1. `1dc61968` — vote-extension epoch-mismatch fix. Proposer-side rejection in `x/checkpointing/prepare/proposal.go`, defense-in-depth re-verify of the BLS multi-sig in `SealCheckpoint`, plus `test/replay/wrong_epoch_ve_test.go`. Closes the gap where a wrong-epoch VE delivered to `LocalLastCommit` as a late precommit could pollute the sealed aggregate.
2. `4d493297` — stake-expansion fixes. Early-unbond-before-inclusion-proof edge case in `x/btcstaking/keeper/msg_server.go`; exports `BtcUndelegate` and `IterateBTCDelegations` wrappers in `x/btcstaking/keeper/btc_delegations.go`; e2ev2 / replay regression tests.

The `release/v4.3.x` version of `4d493297` rewrites `app/upgrades/v4_3/upgrade.go` to embed the phantom stake-expansion remediation. `release/v4.4.x` already shipped its own v4_3 handler (costaking-rewards-tracker reset + epoch boundary validation), so the merge keeps v4.4.x's version untouched. The follow-up commit then realigns the layout (see below).

**Follow-up commits on top of the cherry-picks:**

* `e095c5de` (`feat(upgrades): port phantom stake-expansion remediation to v4_4`) — initial port attempt that put the remediation in a brand-new v4.4 handler.
* `197b32b8` (`docs(btcstaking)`) — flagged `BtcUndelegate` / `IterateBTCDelegations` as v4_4 consumers.
* `dccc33bb` (`refactor(upgrades): align v4_3 with mainnet, move costaking reset to v4_4`) — the layout swap. The v4.3 mainnet binary only ran (1) module migrations and (2) the phantom remediation, so replaying from genesis with v4.4.x's old v4_3 handler would diverge from mainnet state. After this commit:
  * `app/upgrades/v4_3/upgrade.go` is the mainnet phantom-remediation handler, plus a previous-stake covenant-quorum gate (Codex P1): only force-unbond a parent when the child's `PreviousStkCovenantSigs` meet the parent's `CovenantQuorum`, resolved from the parent's own `ParamsVersion`. Without prev-stake quorum the child cannot have produced a valid spending witness on the parent's UTXO, so the parent was not actually spent on Bitcoin and remediation must not touch it.
  * `app/upgrades/v4_4/upgrade.go` is new and holds the costaking-rewards-tracker reset, ActiveBaby zero/recalculate, and epoch boundary validation (the logic that used to live in v4.4.x's v4_3 handler).
  * Phantom-remediation unit tests move to `app/upgrades/v4_3/upgrade_test.go` with a new `TestRemediate_SkipsChildWithoutPreviousStkCovenantQuorum` covering the new gate; costaking-reset test moves to `app/upgrades/v4_4/upgrades_costaking_test.go`.
  * `197b32b8`'s doc updates are reverted because v4.4 no longer consumes `BtcUndelegate` / `IterateBTCDelegations`; the removal TODOs gate on v4.3 retiring alone again.
* `7dac243c` (`chore(e2ev2): retarget upgrade test at v4_4, drop stale comment refs`) — renames `test/e2ev2/upgrades_v4_3_test.go` -> `upgrades_v4_4_test.go` (the costaking ActiveBaby/score recalculation it exercises now lives in v4.4) and retargets it at `v44.UpgradeName`; deletes the superseded `test/e2ev2/upgrades_v4_3_phantom_test.go`; strips `(cometbft#2361)` issue refs from the vote-extension fix comments.
* `c8f4fb19`, `00ba6758` (`chore(ci): GitHub Actions with sha version` / `chore: upgrade GitHub Actions to latest pinned versions`) — cherry-picks of #1995 and #2010 from `main` to pin every action in `.github/workflows/` to its commit SHA. Without these the `check-pinned-actions` job stays red because `release/v4.4.x` was missing the pinning fix from `main`.

### CODE REVIEW

**Comment by RafilxTenfen** [comment]
@codex review

File: app/upgrades/v4_4/upgrade.go [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+			keepers.EncCfg.Codec,
+			coStkStoreService,
+			keepers.EpochingKeeper,
+			keepers.StakingKeeper,
+			keepers.CostakingKeeper,
+		); err != nil {
+			return nil, err
+		}
+
+		return migrations, nil
```
[the remark this answers is not in the document]
Comment (RafilxTenfen) [reply app/upgrades/v4_4/upgrade.go:74]: RemediatePoisonedStakeExpansions was specifically for v43

**Review by Lazar955 — APPROVED** [review]
🫡
